go/vt/sqlparser/token.go

```diff
@@ -460,13 +459,13 @@ func (tkn *Tokenizer) Lex(lval *yySymType) int {
 
 // Error is called by go yacc if there's a parsing error.
 func (tkn *Tokenizer) Error(err string) {
-	buf := &bytes2.Buffer{}
 	if tkn.lastToken != nil {
+		buf := &bytes2.Buffer{}
 		fmt.Fprintf(buf, "%s at position %v near '%s'", err, tkn.Position, tkn.lastToken)
+		tkn.LastError = errors.New(buf.String())
```

Comment:
Since you have a position here, shouldn't you also create a SyntaxError with this message?

Reply:
Done